igor_src/Utils.cpp

```diff
@@ -466,3 +466,53 @@ uint64_t draw_random_64bits_seed(){
 	}
 	return random_seed;
 }
+
+
+UMCodonTable CodonTableStandard = {
+	{"TTT", "F"}, {"TTC", "F"}, {"TTA", "L"}, {"TTG", "L"}, 
+	{"TCT", "S"}, {"TCC", "S"}, {"TCA", "S"}, {"TCG", "S"}, 
+	{"TAT", "Y"}, {"TAC", "Y"}, {"TGT", "C"}, {"TGC", "C"}, 
+	{"TGG", "W"}, {"CTT", "L"}, {"CTC", "L"}, {"CTA", "L"}, 
+	{"CTG", "L"}, {"CCT", "P"}, {"CCC", "P"}, {"CCA", "P"}, 
+	{"CCG", "P"}, {"CAT", "H"}, {"CAC", "H"}, {"CAA", "Q"}, 
+	{"CAG", "Q"}, {"CGT", "R"}, {"CGC", "R"}, {"CGA", "R"}, 
+	{"CGG", "R"}, {"ATT", "I"}, {"ATC", "I"}, {"ATA", "I"}, 
+	{"ATG", "M"}, {"ACT", "T"}, {"ACC", "T"}, {"ACA", "T"}, 
+	{"ACG", "T"}, {"AAT", "N"}, {"AAC", "N"}, {"AAA", "K"}, 
+	{"AAG", "K"}, {"AGT", "S"}, {"AGC", "S"}, {"AGA", "R"}, 
+	{"AGG", "R"}, {"GTT", "V"}, {"GTC", "V"}, {"GTA", "V"}, 
+	{"GTG", "V"}, {"GCT", "A"}, {"GCC", "A"}, {"GCA", "A"}, 
+	{"GCG", "A"}, {"GAT", "D"}, {"GAC", "D"}, {"GAA", "E"}, 
+	{"GAG", "E"}, {"GGT", "G"}, {"GGC", "G"}, {"GGA", "G"}, 
+	{"GGG", "G"},
+	// Stop  codon
+	{"TAA", "*"}, {"TAG", "*"}, {"TGA", "*"},
+	// Start codon
+	{"TTG", "s"}, {"CTG", "s"}, {"ATG", "s"}
+};
+
+
+/*
+ * \param seq: DNA sequence to translate.
+ * \return amino acid sequence. 
+ */
+string translate(string seq){
+	size_t codonPos = 0;
+	const size_t codonLen = 3;
+	size_t seqLen = seq.length();
+	string strCodon = "";
+	string AA ="";
+	string AAChain ="";
+	bool stopCodonQ = false;
+	while ((codonPos < seqLen) and !stopCodonQ){
+		strCodon = seq.substr(codonPos, codonLen);
+//		cout << strCodon << " codonPos: " << codonPos << endl;
+		codonPos = codonPos + codonLen;
```

Comment:
```suggestion
		codonPos += codonPos 
```
prevents from making a copy by directly incrementing the existing object

Reply:
woups I I've just noticed I what I wrote is absurd (take care when I edit on the web), I meant to change for 
```cpp
codonPos += codonLen; 
```
otherwise the code is broken